Rubberduck.Parsing/VBA/DeclarationCaching/DeclarationFinder.cs

```diff
@@ -357,6 +357,17 @@ public ICollection<Declaration> FindFormEventHandlers()
         public IEnumerable<Declaration> Classes => _classes.Value;
         public IEnumerable<Declaration> Projects => _projects.Value;
 
+        /// <summary>
+        /// Gets the <see cref="ProjectDeclaration"/> object for specified referenced project/library.
+        /// </summary>
+        /// <param name="name">The identifier name of the project declaration to find.</param>
+        /// <param name="result">The <see cref="ProjectDeclaration"/> result, if found; null otherwise.</param>
+        public bool TryFindProjectDeclaration(string name, out Declaration result)
+        {
+            result = _projects.Value.SingleOrDefault(project => project.IdentifierName.Equals(name, StringComparison.InvariantCultureIgnoreCase) && !project.IsUserDefined);
```

Comment:
Can we make this `FirstOrDefault` just in case? As it is, this might throw, if somebody names a referenced locked project with a name clash.

Reply:
Ah, good call - I'll also parameterize the `IsUserDefined` check, makes it more versatile. Currently this method has only the two callers it was extracted from.